hr_holidays_public_city/models/hr_holidays_public.py

```diff
@@ -0,0 +1,66 @@
+# Copyright 2023 Tecnativa - Víctor Martínez
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+from odoo import api, fields, models
+
+
+class HrHolidaysPublic(models.Model):
+    _inherit = "hr.holidays.public"
+
+    def _prepare_holidays_list_states_filter_domain(
+        self, pholidays, start_dt, end_dt, employee_id=None
+    ):
+        domain = super()._prepare_holidays_list_states_filter_domain(
+            pholidays=pholidays,
+            start_dt=start_dt,
+            end_dt=end_dt,
+            employee_id=employee_id,
+        )
+        employee = False
+        if employee_id:
+            employee = self.env["hr.employee"].browse(employee_id)
+        if employee and employee.address_id and employee.address_id.city_id:
+            domain += [
+                "|",
+                ("city_ids", "=", False),
+                ("city_ids", "=", employee.address_id.city_id.id),
+            ]
+        else:
+            domain.append(("city_ids", "=", False))
+        return domain
+
+
+class HrHolidaysPublicLine(models.Model):
+    _inherit = "hr.holidays.public.line"
+
+    city_ids = fields.Many2many(
+        "res.city",
+        "hr_holiday_public_city_rel",
+        "line_id",
+        "city_id",
+        "Related Cities",
+    )
+
+    @api.constrains("city_ids")
+    def _check_date_state_city_ids(self):
+        self._check_date_state()
+
+    @api.constrains("city_ids")
+    def _update_calendar_event_city_ids(self):
+        self._update_calendar_event()
+
+    def _prepare_domain_check_date_state_one_state_ids(self):
+        domain = super()._prepare_domain_check_date_state_one_state_ids()
+        if self.city_ids:
+            domain += [("city_ids", "!=", False)]
+        return domain
+
+    def _prepare_domain_check_date_state_one(self):
+        domain = super()._prepare_domain_check_date_state_one()
+        domain += [("city_ids", "=", False)]
+        return domain
+
+    def _prepare_holidays_meeting_values(self):
+        res = super()._prepare_holidays_meeting_values()
+        if self.city_ids:
+            res["description"] += ": " + ", ".join(self.city_ids.mapped("name"))
```

Comment:
It should put the corresponding city of the employee, not the whole city list.

Reply:
Currently this is not possible because the `calendar.event` record is created for each `hr.holidays.public.line` (https://github.com/OCA/hr-holidays/blob/15.0/hr_holidays_public/models/hr_holidays_public.py#L221).

Reply:
OK, my fault. I was thinking this was the specific `hr.leave` corresponding entry.